target/linux/ramips/image/mt7621.mk

```diff
@@ -193,6 +198,27 @@ define Device/dlink_dir-860l-b1
 endef
 TARGET_DEVICES += dlink_dir-860l-b1
 
+define Device/dlink_dir-878-a1
+  $(Device/ubootpad96)
```

Comment:
This one doesn't exist. Note the difference between "Device/" and "Build/" nodes.

Reply:
I'm confused should I have given them different names ?

Reply:
No, it's just that a "Build" definition (e.g. Build/ubootpad96) is something different compared to a "Device" definition (e.g. Device/dlink_dir-878-a1), and have to be used differently.

Reply:
I'm sure I had to add edit mtdsplit_uimage.c to add this device
to find the start of the next available flash block for rootfs

Reply:
will look more into it

Reply:
Well, the Build node makes ubootpad96 available for use below. But there is no Device/ubootpad96 node (which is good, as there is no ubootpad96 device ...), so you cannot include it here.